thespian/system/transport/TCPTransport.py

```diff
@@ -427,11 +426,11 @@ def deadAddress(self, addressManager, childAddr):
         canceli, continuei = partition(lambda i: i[1].targetAddr == childAddr,
                                        self._transmitIntents.items())
         self._transmitIntents = dict(continuei)
-        for each in canceli:
-            each[1].socket.close()
-            delattr(each[1], 'socket')
-            each[1].result = SendStatus.DeadTarget
-            each[1].completionCallback()
+        for _, each in canceli:
```

Comment:
Using a throw-away variable gets rid of the unsightly constant index in the next few lines.